frontend/src/api/task.ts

```diff
@@ -0,0 +1,68 @@
+import { post } from "src/api/requests";
+
+/**
+ * Initializes the shape for the data being retrieved from the backend.
+ */
+export interface Task {
+  _id: string;
+  title: string;
+  description?: string;
+  isChecked: boolean;
+  dateCreated: Date;
+}
+
+interface TaskJSON {
+  _id: string;
+  title: string;
+  description?: string;
+  isChecked: boolean;
+  dateCreated: string;
+}
+
+/**
+ * Converts taks from "any"-typed JSON that only contains primitive types to our
+ * custom Task interface. This is used to change the date field from a string
+ * to a Date object.
+ *
+ * @param task the JSON representation of the task
+ * @returns the typed Task
+ */
+function parseTask(task: TaskJSON): Task {
```

Comment:
```suggestion
/**
 * Converts a Task from JSON that only contains primitive types to our custom
 * Task interface.
 *
 * @param task The JSON representation of the task
 * @returns The parsed Task object
 */
function parseTask(task: TaskJSON): Task {
```

If we were to change how we handle dates, then the comment would be inaccurate; and if we were to add other fields to Task that require special parsing, then the comment would be incomplete. Better to describe what the function does in terms of what its caller cares about: that it correctly transforms a TaskJSON into a Task, by whatever means necessary.